### PR TITLE
Feature/search result

### PR DESCRIPTION
**Pull request by Simon-Duc**

### CODE REVIEW

File: frontend/src/pages/Home.jsx [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+      .get("http://localhost:5001/products/")
+      .then((response) => response.data)
+      .then((data) => {
+        setResults(data);
+      });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.warn(searchValue.current.value);
```
Comment (nibor51) [inline frontend/src/pages/Home.jsx]: delete or comment to justify

**Review by nibor51 — CHANGES_REQUESTED** [review]

File: frontend/src/components/SearchBar.jsx [hunk 2]
```diff
@@ -1,35 +1,61 @@
...
+              placeholder="Recherche ..."
+              ref={searchValue}
+            />
+          </label>
+          <button type="submit">🔎</button>
+        </form>
+      </div>
+      <ul>
+        {results
+          .filter((result) => result.name.includes(searchValue.current.value))
```
Comment (rocambille) [inline frontend/src/components/SearchBar.jsx:47]: the filter should be done on server side

**Review by rocambille — CHANGES_REQUESTED** [review]

File: backend/src/models/AbstractManager.js [hunk 3]
```diff
@@ -14,6 +14,12 @@ class AbstractManager {
     return this.connection.query(`select * from  ${this.table}`);
   }
 
+  findSearch(research) {
+    return this.connection.query(
+      `select * from  ${this.table} WHERE name LIKE '%${research}%' OR brand LIKE '%${research}%'`
```
Comment (rocambille) [inline backend/src/models/AbstractManager.js:19]: congratulations, you just allowed SQL injections :p

https://www.w3schools.com/sql/sql_injection.asp

use markers `?` to inject values in SQL in a secure way

File: backend/src/controllers/ProductController.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ const models = require("../models");
 class ProductController {
   static browse = (req, res) => {
     models.product
-      .findAll()
+      .findSearch(req.query.search)
```
Comment (rocambille) [inline backend/src/controllers/ProductController.js:6]: `search` instead of `findSearch` would be a better name, or `findByQuery`

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by nibor51** [comment]
conflic to resolve

File: frontend/src/components/SearchBar.jsx [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
 
-function SearchBar() {
+export default function SearchBar() {
   const searchValue = useRef();
 
+  const [results, setResults] = useState([]);
+  const getResults = () => {
+    axios
+      .get(
+        `http://localhost:5001/products/?search=${searchValue.current.value}`
```
Comment (rocambille) [inline frontend/src/components/SearchBar.jsx:12]: use env variables instead of magic values ;)

**Review by rocambille — CHANGES_REQUESTED** [review]

**Pull request merged**